src/components/SRNEntry.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <section>
+    <div v-if="plioUse" class="flex flex-col mt-8">
+      <label :class="labelStyleClass">Enter Student SRN</label>
+      <input
+        v-model="singleUserID"
+        type="text"
+        placeholder="Student SRN"
+        required
+        @keypress="isNumberOrLetter($event)"
+        :class="inputStyleClass"
+      />
+      <button @click="this.processForm" :class="buttonStyleClass">SUBMIT</button>
+    </div>
+
+    <div v-else class="flex flex-col mt-8">
+      <label :class="labelStyleClass">Enter Student SRN</label>
+      <div
+        v-for="(input, index) in userIDList"
+        :key="`IDInput-${index}`"
+        :class="inputStyleClass"
+      >
+        <input
+          v-model="input.userID"
+          type="text"
+          placeholder="Student SRN"
+          required
+          @keypress="isNumberOrLetter($event)"
+        />
+        <inline-svg
+          @click="addField(input, userIDList)"
+          class="ml-2 cursor-pointer"
+          :src="require('@/assets/images/add-button.svg')"
+        ></inline-svg>
+        <inline-svg
+          v-show="isAnyUserIDPresent"
+          @click="removeField(index, userIDList)"
+          class="ml-2 cursor-pointer"
+          :src="require('@/assets/images/delete-button.svg')"
+        ></inline-svg>
+      </div>
+      <button :class="buttonStyleClass">SUBMIT</button>
+    </div>
+  </section>
+</template>
+
+<script>
+export default {
+  name: "SRNEntry",
+  props: {
+    redirectTo: String,
+    redirectID: String,
+    purpose: String,
+    subPurpose: String,
+    plioUse: Boolean,
+  },
+  data() {
+    return {
+      userIDList: [{ userID: "" }],
+      singleUserID: "", //is a temp hack
+    };
+  },
+  computed: {
+    buttonStyleClass() {
+      return "block bg-primary hover:bg-primary-hover text-white uppercase text-lg mx-auto p-4 mt-4 rounded";
```

Comment:
I still don't get why this is better than using `@apply` in the styles :D @deepansh96 thoughts?